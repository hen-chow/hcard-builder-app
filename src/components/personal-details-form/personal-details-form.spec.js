import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PersonalDetailsForm from './personal-details-form';

describe('<PersonalDetailsForm />', () => {
  it('invokes the provided onInputChange callback when input changes', () => {
    const onInputChangeCallback = jest.fn();
    const personalDetailsForm = shallow(<PersonalDetailsForm onInputChange={onInputChangeCallback} />);
    
    personalDetailsForm.find('#givenNameInput').simulate('change', { target: { value: 'Hello' } });
    
    expect(onInputChangeCallback.mock.calls.length).toEqual(1);
    expect(onInputChangeCallback.mock.calls[0][0]).toEqual({ target: { value: 'Hello' } });
  });
  
  it('renders the expected HTML', () => {
    const onInputChangeCallback = jest.fn();
    const personalDetailsForm = renderer
      .create(<PersonalDetailsForm onInputChange={onInputChangeCallback} />)
      .toJSON();

    expect(personalDetailsForm).toMatchSnapshot();
  });
});
