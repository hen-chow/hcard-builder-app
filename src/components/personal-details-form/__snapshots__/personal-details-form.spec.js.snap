// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalDetailsForm /> renders the expected HTML 1`] = `
<div
  className="personal-details-form"
>
  <span
    className="form__title"
  >
    Personal details
  </span>
  <hr
    className="line line--small"
  />
  <form>
    <div
      className="form__row"
    >
      <div
        className="form__field form__field--left"
      >
        <label
          htmlFor="givenNameInput"
        >
          Given Name
        </label>
        <input
          id="givenNameInput"
          name="givenName"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
      <div
        className="form__field form__field--right"
      >
        <label
          htmlFor="surnameInput"
        >
          Surname
        </label>
        <input
          id="surnameInput"
          name="surname"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__field form__field--left"
      >
        <label
          htmlFor="emailInput"
        >
          Email
        </label>
        <input
          id="emailInput"
          name="email"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
      <div
        className="form__field form__field--right"
      >
        <label
          htmlFor="phoneInput"
        >
          Phone
        </label>
        <input
          id="phoneInput"
          name="phone"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
    </div>
  </form>
</div>
`;
